### DISCUSSION TITLE
Adding OutputListener to Output Window instances (OutputWriter)

### DISCUSSION THREAD

**Discussion by nedpals** [apache/netbeans#6931] [Q&A]
Hi! I am currently developing a plugin for a [research project](https://github.com/nedpals/bugbuddy) I'm making at the moment which involves listening for incoming error messages from the terminal. My idea was that since Java projects (ant/maven/gradle) run on a dedicated output window, the stderr portion will be passively piped into my program for analysis.

However I find it very difficult to achieve this one (in combination with scattered information and "dead" links which I am frustrated at) to the point that I did the work myself to locate the relevant source code only to find this.

https://github.com/apache/netbeans/blob/94a19f0512b1f1917f474d22e11501ee0f305b76/platform/core.output2/src/org/netbeans/core/output2/ErrWriter.java#L186

Turns out I **can** attach an OutputListener explicitly if my module do the call (eg. `io.getErr().print("test", outputListener)`) but not on this one. Is there anyway I can attach an OutputListener to an `ErrWriter`? Is this monkey-patchable. Hoping for an immediate response. Thanks!

**Comment by Chris2011** [comment]
Hey @nedpals sry that you didn't get any response. I can't answer this question but maybe someone else @neilcsmith-net, @mbien, @sdedic ?
